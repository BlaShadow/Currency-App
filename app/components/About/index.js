import AboutComponent from './AboutComponent'
import Styles from './style'

export { AboutComponent, Styles } 