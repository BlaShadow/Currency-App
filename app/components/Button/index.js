import ClearButton from './ClearButton';
import styles from './style'; 

export {ClearButton, styles};