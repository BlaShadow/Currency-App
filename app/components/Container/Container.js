import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { View, TouchableWithoutFeedback, Keyboard } from 'react-native';

import styles from './styles';

class Container extends React.PureComponent {
    render(){
        const { children, color } = this.props;
        const containerStyles = [styles.container];

        containerStyles.push({backgroundColor: color});
        
        return (
            <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
                <View style={containerStyles}>
                    {children}
                </View>
            </TouchableWithoutFeedback>
        );
    }
}

Container.propTypes = {
    children: PropTypes.any,
};

const stateToProps = (state) => {
    return {
        color: state.appState.mainColor
    };
};

export default connect(stateToProps)(Container);

