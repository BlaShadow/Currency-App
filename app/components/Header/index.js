import Header from './Header';
import style from './style';

export { Header, style };