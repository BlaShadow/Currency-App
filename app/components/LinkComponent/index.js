import { LinkComponent } from './LinkComponent'

export { LinkComponent }