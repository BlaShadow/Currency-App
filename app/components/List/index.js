import ListItem from './ListItem';
import Separator from './separator';

export { ListItem, Separator };