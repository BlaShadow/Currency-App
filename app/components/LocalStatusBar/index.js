import LocalStatusBar from './LocalStatusBar';

export { LocalStatusBar };