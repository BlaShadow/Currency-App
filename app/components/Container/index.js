import Container, { Container as ContainerComponent } from './Container';
import styles from './styles';  

export { Container, styles, ContainerComponent };