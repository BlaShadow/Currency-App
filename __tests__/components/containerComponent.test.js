import 'react-native';

import React from 'react'
import { ContainerComponent } from '../../app/components/Container'

import renderer from 'react-test-renderer';

test('Should render Container correctly', () => {
    const tree = renderer.create(<ContainerComponent />).toJSON();

    expect(tree).toMatchSnapshot();
});