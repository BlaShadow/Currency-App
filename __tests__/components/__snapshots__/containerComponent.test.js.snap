// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Container correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
  testID={undefined}
/>
`;
